Framework/DataObjects/src/EventList.cpp

```diff
@@ -1567,6 +1632,93 @@ void EventList::compressEventsParallelHelper(
+  out.clear();
+  // We will make a starting guess of 1/20th of the number of input events.
+  out.reserve(events.size() / 20);
+
+  // The last TOF to which we are comparing.
+  double lastTof = std::numeric_limits<double>::lowest();
+  // For getting an accurate average TOF
+  double totalTof = 0;
+
+  // pulsetime bin information - stored as int nanoseconds because it
+  // implementation type for DateAndTime object
+  const int64_t pulsetimeStart = timeStart.totalNanoseconds();
+  const int64_t pulsetimeDelta = static_cast<int64_t>(seconds * SEC_TO_NANO);
+
+  // pulsetime information
+  std::vector<DateAndTime> pulsetimes; // all the times for new event
+
+  // Carrying weight and error
+  double weight = 0;
+  double errorSquared = 0;
+
+  // move up to first event that has a large enough pusletime
+  auto it = events.cbegin();
+  for (; it != events.cend(); ++it) {
+    if (it->m_pulsetime >= timeStart)
+      break;
+  }
+
+  // bin if the pulses are histogrammed
+  int64_t lastPulseBin =
+      (it->m_pulsetime.totalNanoseconds() - pulsetimeStart) / pulsetimeDelta;
+  // loop through events and accumulate weight
+  for (; it != events.cend(); ++it) {
+    const int64_t eventPulseBin =
+        (it->m_pulsetime.totalNanoseconds() - pulsetimeStart) / pulsetimeDelta;
+    if ((eventPulseBin <= lastPulseBin) &&
+        (std::fabs(it->m_tof - lastTof) <= tolerance)) {
+      // Carry the error and weight
+      weight += it->weight();
+      errorSquared += it->errorSquared();
+      // Track the average tof
+      totalTof += it->m_tof;
+      // Accumulate the pulse times
+      pulsetimes.push_back(it->m_pulsetime);
+    } else {
+      // We exceeded the tolerance
+      if (!pulsetimes.empty()) {
+        // Create a new event with the average TOF and summed weights and
+        // squared errors.
+        out.emplace_back(totalTof / static_cast<double>(pulsetimes.size()),
+                         Kernel::DateAndTimeHelpers::averageSorted(pulsetimes),
+                         weight, errorSquared);
+      }
+      // Start a new combined object
+      totalTof = it->m_tof;
+      weight = it->weight();
+      errorSquared = it->errorSquared();
+      lastTof = it->m_tof;
+      lastPulseBin = eventPulseBin;
+      pulsetimes.clear();
+      pulsetimes.push_back(it->m_pulsetime);
+    }
+  }
+
+  // Put the last event in there too.
+  if (!pulsetimes.empty()) {
+    // Create a new event with the average TOF and summed weights and squared
+    // errors.
+    out.emplace_back(totalTof / static_cast<double>(pulsetimes.size()),
+                     Kernel::DateAndTimeHelpers::averageSorted(pulsetimes),
+                     weight, errorSquared);
+  }
+
+  // If you have over-allocated by more than 5%, reduce the size.
+  size_t excess_limit = out.size() / 20;
+  if ((out.capacity() - out.size()) > excess_limit) {
+    // Note: This forces a copy!
+    std::vector<WeightedEvent>(out).swap(out);
+  }
```

Comment:
How about `std::vector::shrink_to_fit()`?

Reply:
Thanks for the pointer. The original implementation of `EventList` was before we used C++11 and I didn't know of this "new" method.